libvast/vast/system/index_client_actor.hpp

```diff
@@ -19,11 +19,17 @@
 
 namespace vast::system {
 
+/// The PARTITION CLIENT actor interface.
+// FIXME: Move into separate header.
+using partition_client_actor = caf::typed_actor<caf::reacts_to<ids>>;
+
 /// The INDEX CLIENT actor interface.
 using index_client_actor = caf::typed_actor<
   // Receives ids from the INDEX for partial query hits.
-  caf::reacts_to<ids>,
+  // caf::reacts_to<ids>,
   // Receives done from the INDEX when the query finished.
-  caf::reacts_to<atom::done>>;
+  caf::reacts_to<atom::done>>
+  // FIXME: comment
+  ::extend_with<partition_client_actor>;
```

Comment:
Do we actually need *both*?

Reply:
The alternative would be to implement a dummy `done` handler for the partition clients that never gets called. Which is extra confusing because the reader mustn't confuse it with the "outer" `done`-handler that handles the end of the request.

Reply:
I think I understand it now. It is required because client gets the `done` from the `query_supervisor`. I think we should change the evaluator to not send partial results anymore soon, then we can avoid this complex multi-party protocol.

Reply:
Are you sure about this? The measurements from #1220 have shown that the partial results have quite the impact. If we only send a `vector<ids>` instead of using hand-rolled streaming of `ids` then we lose a fair bit of performance.

I think this should be refactored to use CAF streaming instead of our poor man's version of it.

Reply:
Either way, I'd suggest doing that as a separate refactoring.

Reply:
(also, @tobim, that's almost correct but the partition itself also sends a `done` atom [see partition_actor.hpp], it's just that this is sent as a response to the expression request and is therefore not part of the callback interface)